src/api/init.ts

```diff
@@ -0,0 +1,52 @@
+import Fastify, { FastifyBaseLogger, FastifyInstance } from 'fastify';
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
+import { FtRoutes } from './routes/ft';
+import { NftRoutes } from './routes/nft';
+import { PgStore } from '../pg/pg-store';
+import { ENV } from '../env';
+import FastifyCors from '@fastify/cors';
+import FastifySwagger from '@fastify/swagger';
+import { StatusRoutes } from './routes/status';
+
+export function buildApiServer(args: { db: PgStore }) {
+  const fastify = Fastify({
+    trustProxy: true,
+    logger: true,
+  }).withTypeProvider<TypeBoxTypeProvider>();
+
+  fastify.decorate('db', args.db);
+
+  fastify.register(FastifyCors);
```

Comment:
For whatever reason, Fastify v4 requires these `register` calls to be async now https://www.fastify.io/docs/latest/Guides/Migration-Guide-V4/#synchronous-route-definitions

```suggestion
  await fastify.register(FastifyCors);
```

I think it only matters depending on the route/plugin behavior, but probably safer to await all of them. The docs mention better error handling.